File: Task4/JavaBenchmark/src/BenchmarkCalculation/Calculations.java
package BenchmarkCalculation;

public interface Calculations<M> {
    public abstract M[][] matrixTransposing(M[][] matrix);
    public abstract M[][] matrixMultiplication(M[][] matrixA, M[][] matrixB);
    public abstract M[][] scalarMultiplication(M[][] matrix, M scalar);
    public abstract M[][] matrixSum(M[][] matrixA, M[][] matrixB);
}

File: Task4/CSharpBenchmark/Calculations.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CSharp
{
    internal abstract class Calculations<M>
    {
        public abstract M[,] MatrixTransposing(M[,] matrix);
        public abstract M[,] MatrixMultiplication(M[,] matrixA, M[,] matrixB);
        public abstract M[,] ScalarMultiplication(M[,] matrix, M scalar);
        public abstract M[,] MatrixSumm(M[,] matrixA, M[,] matrixB);
        public abstract M[,] MatrixMultiplicationThreads(M[,] matrixA, M[,] matrixB, int threads);
    }
}
